docs/design-documents/README.md

```diff
@@ -0,0 +1,44 @@
+
+## Design Process
+* Developer generates a design document located under *https://github.com/ARMmbed/mbed-os/docs/design-documents/{AREA OR COMPONENT}* and generates a pull-request to Mbed-OS.
+  * NOTE: If you are part of an organization and would like to do design review within your organization before landing the pull-request, its ok to do so.
+  * The pull request may also include the initial code implementation. It's upto to the develper to decide if the initial implementation comes along with the design document or later.
+  * Although, if you are planning a large change, its encoraged to start the design review early before the implementation. This also help to avoid major re-work of your implementation if required by the design review.
+* Design review follows the same process as code review. Please refer contribution process as outlined in [Contributing to Mbed-OS](https://os.mbed.com/contributing/) for more information.
+* Once the design is approved, implementation is completed/adjusted according to the latest design document. Developer can create the pull-request with the final implementation, if its coming separately.
+
+## Capturing future changes to design
+If your implementation changes in future, corresponsding design documents may be updated to capture the new design.
+It's possible that a new implementation may require changes to some previous implementations. In that case, capture
+the new changes in the original design document as well. Also create references to other(or older) design documents in the
+References section for the new design document.
+
+## General guidelines for creating design documents
+Its up to the developer to make a judgement on whether a design docucment is required for a change. It may not be required for minor fixes and small localized changes. But for changes with lot of code turmoil across multiple components or if its new feature or functionality being added its best to have a backing design document. Its also ok for developers to gather initial feedback on some idea before they decide to implement. Its also not absolutely necessary that implementation should be started only after design document review. The caveat here is if the proposed design is changed significantly in the process of review it will impact your implementation as well. Thus for large features where lot of design feedbacks are expected it's advisable to start your implementation once major feedbacks are captured. For example, API designs can invite lot of feedbacks. In such cases, its best to capture those feedbacks before starting the implementation cycle. But for very localized implementations not impacting other parts of the system/APIs it may be ok to provide an initial implementation(in the same PR) as a reference along with design document.
```

Comment:
"docucment" / "its" -> "it's" / "feedbacks" -> "feedback"

Reply:
Is it a summary? It feels a bit redundant with what's explained in the paragraph starting line 4. Might be worth breaking this down in bullet points (here are the steps).